docs: add Claude Code and Codex installation and usage instructions (#139)
gemini-cli-extensions/cloud-sql-postgresql: README.md

This repository provides a set of agent skills to interact with [Cloud SQL for PostgreSQL](https://cloud.google.com/sql/docs/postgres) instances. These skills can be used with various AI agents, including [Gemini CLI](https://google-gemini.github.io/gemini-cli/), Claude Code, and Codex, to manage your databases, execute queries, explore schemas, and troubleshoot issues using natural language prompts.

Learn more about [Gemini CLI Extensions](https://github.com/google-gemini/gemini-cli/blob/main/docs/extensions/index.md).
> [!IMPORTANT]
> **We Want Your Feedback!**
> Please share your thoughts with us by filling out our feedback [form][form]. 
> Please share your thoughts with us by filling out our feedback [form][form].
> Your input is invaluable and helps us improve the project for everyone.

[form]: https://docs.google.com/forms/d/e/1FAIpQLSfEGmLR46iipyNTgwTmIDJqzkAwDPXxbocpXpUbHXydiN1RTw/viewform?usp=pp_url&entry.157487=cloud-sql-postgresql

> ## Why Use the Cloud SQL for PostgreSQL Extension?

Author: Added a TOC

## Prerequisites

Before you begin, ensure you have the following:

> - An AI agent installed (e.g., [Gemini CLI](https://github.com/google-gemini/gemini-cli) version **+v0.6.0**, [Claude Code](https://claude.ai/install.sh), or [Codex](https://openai.com/blog/openai-codex/)).

Author: Added specific versions. Thanks for pointing out.

- A Google Cloud project with the **Cloud SQL Admin API** enabled.
- Ensure [Application Default Credentials](https://cloud.google.com/docs/authentication/gcloud) are available in your environment.
- IAM Permissions:
  - Cloud SQL Client (`roles/cloudsql.client`)
  - Cloud SQL Admin (`roles/cloudsql.admin`)

> [!NOTE]
> If you do not configure a specific `CLOUD_SQL_POSTGRES_USER` or `CLOUD_SQL_POSTGRES_PASSWORD`, these skills default to using the active local IAM user credentials. You must also add the IAM user to your Cloud SQL instance, see [Creating a database user](https://cloud.google.com/sql/docs/postgres/add-manage-iam-users#creating-a-database-user).

## Getting Started

### Installation & Usage

To start interacting with your database, install the skills for your preferred AI agent, then launch the agent and use natural language to ask questions or perform tasks.

#### Gemini CLI

> **1. Install the extension:**

Author: Updated.

gemini extensions install https://github.com/gemini-cli-extensions/cloud-sql-postgresql
```

**2. (Optional) Manage Configuration:**
To view or update your configuration in Gemini CLI:
- Terminal: `gemini extensions config cloud-sql-postgresql [setting name] [--scope <scope>]`
- Gemini CLI: `/extensions list`

**3. Start the agent:**
```bash
gemini
```
_(Tip: Run `/extensions list` to verify your configuration and active extensions.)_

#### Claude Code

**1. Start the agent:**
```bash
claude
```

**2. Add the marketplace:**
```bash
/plugin marketplace add gemini-cli-extensions/cloud-sql-postgresql
```

**3. Install the plugin:**
```bash
/plugin install cloud-sql-postgresql@google-data-cloud-skills
```
_(Tip: Run `/plugin list` inside Claude Code to verify the plugin is active, or `/reload-plugins` if you just installed it.)_

#### Codex

**1. Clone the Repo:**
```bash
git clone git@github.com:gemini-cli-extensions/cloud-sql-postgresql.git
```

**2. Install the plugin:**
```bash
mkdir -p ~/.codex/plugins
cp -R /absolute/path/to/cloud-sql-postgresql ~/.codex/plugins/cloud-sql-postgresql
```

**3. Create or update marketplace.json:**
`~/.agents/plugins/marketplace.json`
```json
{

>   "name": "google-data-cloud-skills",

Reviewer: ```suggestion
  "name": "my-google-data-cloud-marketplace",
```
I think this would be nice for everyone to use a as personal. We will use another name for the official repo.

  "interface": {
    "displayName": "Google Data Cloud Skills"
  },
  "plugins": [
    {
      "name": "cloud-sql-postgresql",
      "source": {
        "source": "local",
        "path": "./plugins/cloud-sql-postgresql"
      },
      "policy": {
        "installation": "AVAILABLE",
        "authentication": "ON_INSTALL"
      },
      "category": "Database"
    }
  ]
}

```

_(Tip: Run `codex plugin list` or use the `/plugins` interactive menu to verify your installed plugins.)_

> ### Configuration

Author: I have moved it above install and clarified the correct steps for env var setting. Let me know if you think this makes sense.

> You may be prompted to configure the following settings during installation. These settings can also be set as environment variables.

Author: I have added proper step by step instructions for this. The addition of vars currently varies for both. In the future, we will try to unify the experience.

Please let me know if it's unclear

# Cloud SQL for PostgreSQL Agent Skills

> [!NOTE]

> > These skills are currently in beta (pre-v1.0), and may see breaking changes until the first stable release (v1.0).

Reviewer: ```suggestion
> This extension/plugin is currently in beta (pre-v1.0), and may see breaking changes until the first stable release (v1.0).
```

Reviewer: not sure if I like my suggestion

Author: Updated it with a better idea. Let me know what you think!

</details>

<details>
<summary id="codex">Codex</summary>

**1. Clone the Repo:**

```bash
git clone --branch 0.3.0 git@github.com:gemini-cli-extensions/cloud-sql-postgresql.git
```

**2. Install the plugin:**

```bash
mkdir -p ~/.codex/plugins
cp -R /absolute/path/to/cloud-sql-postgresql ~/.codex/plugins/cloud-sql-postgresql
```

**3. Set env vars:**
Enter your environment vars as described in the [configuration section](#configuration).

**4. Create or update marketplace.json:**
`~/.agents/plugins/marketplace.json`

```json
{
  "name": "my-data-cloud-google-marketplace",
  "interface": {
    "displayName": "Google Data Cloud Skills"
  },
  "plugins": [
    {
      "name": "cloud-sql-postgresql",
      "source": {
        "source": "local",
        "path": "./plugins/cloud-sql-postgresql"
      },
      "policy": {
        "installation": "AVAILABLE",
        "authentication": "ON_INSTALL"
      },
      "category": "Database"
    }
  ]
}
```

_(Tip: Run `codex plugin list` or use the `/plugins` interactive menu to verify your installed plugins.)_

</details>

<!-- {x-release-please-end} -->

> > [!WARNING]

Reviewer: Remove this or move under Gemini CLI

Author: Moved. Thanks for pointing out.

- [Cloud SQL for PostgreSQL Replication](./skills/cloud-sql-postgres-replication/SKILL.md) - Use these skills when you need to monitor replication health, manage sync states between nodes, and audit database roles and security settings to ensure environment integrity.
- [Cloud SQL for PostgreSQL View Config](./skills/cloud-sql-postgres-view-config/SKILL.md) - Use these skills when you need to discover and manage PostgreSQL extensions or fine-tune engine-level settings such as memory allocation and server configuration parameters.

## Additional Agent Skills

Find additional skills to support your entire software development lifecycle at [github.com/gemini-cli-extensions](https://github.com/gemini-cli-extensions), including:

- [Generic PostgreSQL skills](https://github.com/gemini-cli-extensions/postgres)
- [Cloud SQL for PostgreSQL Observability skills](https://github.com/gemini-cli-extensions/cloud-sql-postgresql-observability)
- and more!

> ## Troubleshooting

Reviewer: nit: let's clean these up in the future

Author: Kept a note of it.
